src/storage/src/source/mysql/replication.rs

```diff
@@ -308,10 +308,12 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
                 metrics.total.inc();
 
                 match event_data {
-                    Some(EventData::HeartbeatEvent) => {
+                    Some(EventData::HeartbeatEvent) | Some(EventData::XidEvent(_)) => {
```

Comment:
The assumption that a heartbeat only arrives at transaction boundaries is not documented anywhere and has always been some amount of scary to me. Now that we found the xid event we should not be doing anything related with capabilities when a heartbeat event arrives so I would suggest changing this to just `Some(EventData::XidEvent(_))`.

Reply:
That's a great point!

Reply:
Okay fair point, will remove the hearbeat!

Reply:
Okay without the heartbeat we now don't advance the frontier if the last event received is a non-transaction event (DDL changes / query-events). So the tests are currently failing when we are expecting an error after some DDL events.

We could reintroduce the hearbeat for this, or we could advance the frontier after each DDL statement, which I think is "correct" but is another unspecified behavior we'd be relying on.

Reply:
I've pushed a second commit with the approach above -- moving the frontier forward on DDL changes and explicit COMMIT statements too.